### PR TITLE
HLM-4064: Added adverse event popup, updated localisation,updated bloc for futuretask

### PR DESCRIPTION
**Pull request by anilsingha-eGov**

### CODE REVIEW

File: apps/health_campaign_field_worker_app/lib/blocs/delivery_intervention/deliver_intervention.dart [hunk 1]
```diff
@@ -82,11 +82,12 @@ class DeliverInterventionBloc
...
           .where((element) =>
               element.additionalFields?.fields
-                  .firstWhere(
-                    (a) => a.key == "DeliveryStrategy",
-                  )
-                  .value ==
-              DeliverStrategyType.indirect.name.toUpperCase())
+                      .firstWhere(
+                        (a) => a.key == "DeliveryStrategy",
+                      )
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/blocs/delivery_intervention/deliver_intervention.dart:87]: Add to  enum

File: apps/health_campaign_field_worker_app/lib/pages/beneficiary/beneficiary_details.dart [hunk 2]
```diff
@@ -103,8 +103,8 @@ class _BeneficiaryDetailsPageState
                   ? int.tryParse(
                         lastDose,
                       ) ??
-                      1
-                  : 1,
+                      0
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/pages/beneficiary/beneficiary_details.dart:106]: This should be 1 right

File: apps/health_campaign_field_worker_app/lib/pages/beneficiary/record_past_delivery_details.dart [hunk 3]
```diff
@@ -65,21 +67,71 @@ class _RecordPastDeliveryDetailsPageState
...
+                              final parent = router.parent() as StackRouter;
+                              router.pop();
+
+                              Navigator.of(
+                                ctx,
+                                rootNavigator: true,
+                              ).pop();
+                              router.push(
+                                AdverseEventsRoute(
+                                  tasks: [state.futureTask!.last],
```
Comment (naveen-egov) [inline apps/health_campaign_field_worker_app/lib/pages/beneficiary/record_past_delivery_details.dart:102]: Add the null check

**Review by naveen-egov — CHANGES_REQUESTED** [review]

File: apps/health_campaign_field_worker_app/lib/blocs/delivery_intervention/deliver_intervention.dart (hunk 1 above)
Comment (anilsingha-eGov) [reply apps/health_campaign_field_worker_app/lib/blocs/delivery_intervention/deliver_intervention.dart:90]: Added enum

**Review by naveen-egov — APPROVED** [review]

**Pull request merged**